store/src/main/scala/com/improving/app/store/domain/Store.scala

```diff
@@ -0,0 +1,401 @@
+
+    if (maybeValidationError.isDefined) {
+      Left(maybeValidationError.get)
+    } else {
+      val newMetaInfo = updateMetaInfo(metaInfo = state.metaInfo, lastUpdatedByOpt = command.onBehalfOf)
+      val info: Either[ValidationError, StoreInfo] = state match {
+        case x: DraftState =>
+          val draftInfo = x.info.getOrElse(EditableStoreInfo.defaultInstance)
+          val newInfo =
+            StoreInfo(draftInfo.getName, draftInfo.getDescription, draftInfo.sponsoringOrg)
+          val maybeInfoValidationError: Option[ValidationError] = createdStateStoreInfoValidator(newInfo)
+          if (maybeInfoValidationError.isDefined) {
+            Left(maybeValidationError.get)
+          } else {
+            Right(newInfo)
+          }
+        case x: ClosedState => Right(x.info)
+      }
+
+      if (info.isLeft) {
+        Left(info.left.getOrElse(ValidationError.apply("Unknown validation error")))
+      } else {
+
+        Right(
+          StoreDeleted(
+            storeId = command.storeId,
+            info = Some(info.getOrElse(StoreInfo.defaultInstance)),
+            metaInfo = Some(newMetaInfo)
+          )
+        )
+      }
+    }
+  }
+
+  private def terminateStore(state: InitializedState, terminate: TerminateStore): Either[Error, StoreEvent] = {
+    val maybeValidationError: Option[ValidationError] = applyAllValidators[TerminateStore](
+      Seq(
+        c => required("store id", storeIdValidator)(c.storeId),
+        c => required("on behalf of", memberIdValidator)(c.onBehalfOf)
+      )
+    )(terminate)
+
+    if (maybeValidationError.isDefined) {
+      Left(maybeValidationError.get)
+    } else {
+      val newMetaInfo = updateMetaInfo(metaInfo = state.metaInfo, lastUpdatedByOpt = terminate.onBehalfOf)
+      Right(
+        StoreTerminated(
+          storeId = terminate.storeId,
+          metaInfo = Some(newMetaInfo)
+        )
+      )
+    }
+  }
+
+  private def editStoreInfo(
+      state: InitializedState,
+      command: EditStoreInfo
+  ): Either[Error, StoreInfoEdited] = {
+    val maybeValidationError = applyAllValidators[EditStoreInfo](
+      Seq(
+        command => required("store id", storeIdValidator)(command.storeId),
+        command => required("on behalf of", memberIdValidator)(command.onBehalfOf),
+      )
+    )(command)
+    if (maybeValidationError.isDefined) {
+      Left(maybeValidationError.get)
+    } else {
+      val stateInfo: Either[ValidationError, StoreInfo] = state match {
+        case x: DraftState =>
+          val draftInfo = x.info.getOrElse(EditableStoreInfo.defaultInstance)
+          Right(StoreInfo(draftInfo.getName, draftInfo.getDescription, draftInfo.sponsoringOrg))
+        case x: CreatedState => Right(x.info)
+        case _               => Left(ValidationError.apply("Editing is not allowed in this state"))
+      }
+
+      if (stateInfo.isLeft) {
+        Left(stateInfo.left.getOrElse(ValidationError.apply("Unknown validation error")))
```

Comment:
I don't think this else case is needed

Reply:
use `.get` instead: the else case should never happen

I don't believe this will go away because not all validation will be able to be moved. For example: id validation will need to check that the corresponding entity actually exists, not merely that it is syntactically correct.

Reply:
Would love to if `.get` was not deprecated

<img width="172" alt="Screenshot 2023-05-08 at 2 46 05 PM" src="https://user-images.githubusercontent.com/105220856/236906478-17e1f0bb-15e3-4eb9-9b39-0f26722f601c.png">

Reply:
In protobuf we can make sure that the string is non-empty, so I'm not sure why we would have to put that validation here. Maybe it would make us feel safer, but it wouldn't be necessary given protobuf validation not letting in empty strings. In EditableInfo a None will be a 'dont edit' while a Some("") will be an error (invalid edit)

Reply:
What about just returning from the state match and having a helper method for the success cases?

Reply:
Pushed the changed - looks much nicer :)